src/OSPSuite.Presentation/Presenters/Importer/ImporterPresenter.cs

```diff
@@ -172,26 +174,14 @@ private void loadSheetsFromDataPresenter(object sender, ImportSheetsEventArgs ar
          {
             loadSheets(args.DataSourceFile, args.Sheets, args.Filter);
             _importerDataPresenter.DisableImportedSheets();
-            foreach (var sheet in args.Sheets.Keys)
-               _configuration.AddToLoadedSheets(sheet);
-
+            args.Sheets.Keys.Each(_configuration.AddToLoadedSheets);
             _configuration.FilterString = args.Filter;
          }
          catch (Exception e)
          {
-            {
-               _dialogCreator.MessageBoxError
-               (
-                  e is AbstractImporterExceptions ? 
-                  e.Message : 
-                  Captions.Importer.UnexpectedExceptionWhenLoading
-               );
-               
-               foreach (var sheetName in args.Sheets.Keys)
-               {
-                  _importerDataPresenter.Sheets.Remove(sheetName);
-               }
-            }
+            var message = e is AbstractImporterException ? e.Message : Captions.Importer.UnexpectedExceptionWhenLoading;
+            _dialogCreator.MessageBoxError(message);
+            args.Sheets.Keys.Each(_importerDataPresenter.Sheets.Remove);
```

Comment:
It's good to use a more functional way of writing for loops in my option. I'd suggest to avoid foreach in general